Description: Rename variable async to async_watcher
Author: Andrew Olefira <andrew.olefira@gmail.com>
Date: Thu, 18 Jan 2018 19:29:11 +0200
Origin: upstream, https://codecov.io/gh/pika/pika/commit/3d60e5415a24c5d59fa1fbb4c947e0c42c75c2df
Last-Update: 2018-07-10

Index: python-pika/pika/adapters/libev_connection.py
===================================================================
--- python-pika.orig/adapters/libev_connection.py
+++ python-pika/pika/adapters/libev_connection.py
@@ -102,7 +102,7 @@ class LibevConnection(BaseConnection):
                 self.ioloop = pyev.default_loop()
                 self.ioloop.update()
 
-        self.async = None
+        self.async_watcher = None
         self._on_signal_callback = on_signal_callback
         self._io_watcher = None
         self._active_timers = {}
@@ -143,8 +143,8 @@ class LibevConnection(BaseConnection):
 
             # NOTE: if someone knows why this async is needed here, please add
             # a comment in the code that explains it.
-            self.async = pyev.Async(self.ioloop, self._noop_callable)
-            self.async.start()
+            self.async_watcher = pyev.Async(self.ioloop, self._noop_callable)
+            self.async_watcher.start()
 
             if self._on_signal_callback:
                 global_sigterm_watcher.start()
